src/observer/index.js

```diff
@@ -182,18 +198,18 @@ function defineReactive (obj, key, val) {
         if (childOb) {
           childOb.dep.depend()
         }
-        if (_.isArray(val)) {
-          for (var e, i = 0, l = val.length; i < l; i++) {
-            e = val[i]
+        if (_.isArray(target.val)) {
```

Comment:
Since `val` is now a getter, if the original is also a getter, we are still effectively invoking the original getter multiple times here. We should just cache the value as `var value = target.val` at the top of the metaGetter.